search-processors/src/main/java/org/opensearch/searchpipelines/questionanswering/generative/llm/LlmIOUtil.java

```diff
@@ -57,7 +59,19 @@ public static ChatCompletionInput createChatCompletionInput(
         List<String> contexts,
         int timeoutInSeconds
     ) {
-
-        return new ChatCompletionInput(llmModel, question, chatHistory, contexts, timeoutInSeconds, systemPrompt, userInstructions);
+        Llm.ModelProvider provider = Llm.ModelProvider.OPENAI;
```

Comment:
Do you have plan to make this solution generic for the LLMs? Or we should have separate handler for every LLMs? May be we can add a todo consider how can we make this class more generic to all LLMs?

Reply:
#1475 - this is what I am proposing.